scala-thoughts.md

```diff
@@ -0,0 +1,39 @@
+The first thing I would say about Scala is that there is too much syntax. It
+seems that there are several different ways of doing every simple thing, and
+which syntax you should use in each place is very unclear. For example, I'm
+still unclear what the difference between the following statements are:
+
+    f g x
+    f.g(x)
+    f(g(x))
+
+Since Scala likes implementing DSLs so much, it appears that lots of packages
+are more DSL than API, which usually makes the API easier to read, but harder
+to learn and write since you have to learn a largely new syntax over again.
+
+Scala is really cool in that it can be used in lots of different ways and each
+one is pretty nice to use. C++ also aims to be able to be used in many
+different ways, but its syntax gets really obscene for things like functional
+programming because it basically runs out of short syntax and thus has to go
+into longer, more convoluted things. Scala has so much syntax that it can
+implement all these different programming models without getting too gross.
+Scala is still weighted toward the functional model but the object-oriented
+model is nice and imperative programming is... possible.
+
```

Comment:
definitely not how it was intended to be used, but you could kinda just write java (with a few clear exceptions)